.github/workflows/cloudflare-pages.yml

```diff
@@ -0,0 +1,92 @@
+# NOTE: This name appears in GitHub's Checks API and in workflow's status badge.
+name: cloudflare-pages
+
+# Trigger the workflow when:
+on:
+  # A push occurs to one of the matched branches.
+  push:
+    branches: [master]
+  # Or when a pull request event occurs for a pull request against one of the
+  # matched branches.
+  pull_request:
+    branches: [master]
+
+jobs:
+  publish:
+    # This name appears in GitHub's Checks API.
+    name: cloudflare-pages
+    # Do not trigger job for dependency update bot.
+    if: github.actor != 'renovate[bot]'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      # In progress comment will be updated with a deploy status.
+      - name: Add in progress comment to PR
+        uses: mshick/add-pr-comment@v2
+        with:
+          message: |
+            ## Deploying with <a href="https://pages.dev">Cloudflare Pages</a>
+            <table>
+              <tr>
+                <td><strong>Latest commit:</strong></td>
+                <td><code>${{ github.event.pull_request.head.sha }}</code></td>
+              </tr>
+              <tr>
+                <td><strong>Status:</strong></td><td>⏳ In progress</td>
+              </tr>
+            </table>
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Node.js 18
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: yarn
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Build app preview
+        run: yarn build-preview
+      - name: Publish to Cloudflare Pages
+        # Id is needed to access output in a next step.
+        id: cloudflare
+        uses: cloudflare/pages-action@v1
+        with:
+          apiToken: ${{ secrets.CLOUDFRONT_API_TOKEN }}
```

Comment:
I've used `CLOUDFLARE_PAGES_API_TOKEN` and `CLOUDFLARE_ACCOUNT_ID`.

Please, update the code accordingly.